bin/src/hickory-dns.rs

```diff
@@ -351,6 +360,15 @@ fn run() -> Result<(), String> {
         return Err("dropping privileges is only supported on Unix systems".to_string());
     }
 
+    #[cfg(unix)]
+    {
+        let token = server.shutdown_token();
+        task::spawn(async move {
```

Comment:
I think it would be more idiomatic to spell this as `tokio::spawn()` (and save an import).